invokeai/app/invocations/denoise_latents.py

```diff
@@ -707,9 +714,110 @@ def prepare_noise_and_latents(
 
         return seed, noise, latents
 
+    def invoke(self, context: InvocationContext) -> LatentsOutput:
+        if os.environ.get("USE_MODULAR_DENOISE", False):
+            return self._new_invoke(context)
+        else:
+            return self._old_invoke(context)
+
     @torch.no_grad()
     @SilenceWarnings()  # This quenches the NSFW nag from diffusers.
-    def invoke(self, context: InvocationContext) -> LatentsOutput:
+    def _new_invoke(self, context: InvocationContext) -> LatentsOutput:
+        # TODO: remove supression when extensions which use models added
+        with ExitStack() as exit_stack:  # noqa: F841
```

Comment:
Can we remove this until its needed? I'm note quite sure how you intend to use it, but I'm doubtful it'll need such a long lifetime.

Reply:
I feel that lifetime probably ok, it just feel big due to definition so early(and latents/noise/scheduler/conditionings simply not use it).
In PR it used by extensions which load models, so for now I think I'll remove it.